js/style/style.js

```diff
@@ -420,7 +399,7 @@ Style.prototype = util.inherit(Evented, {
         }
         this._validateLayer(layer);
 
-        layer.on('error', this._forwardLayerEvent);
+        layer.setEventedParent(this, {layer: {id: layer.id}});
```

Comment:
As this is a public event, I suspect that changing the structure would impact our users.

The intent behind this structure is to give us the flexibility to make `StyeLayer`s part of the public API. Think of `{id: ...}` as the public interface of `StyleLayer`. 😅 

I am working on adding custom layer types now, which bring these classes further into the public API.